pandas/core/index.py

```diff
@@ -1649,15 +1654,20 @@ def get_indexer(self, target, method=None, limit=None):
         Parameters
         ----------
         target : Index
-        method : {None, 'pad'/'ffill', 'backfill'/'bfill', 'nearest'}
+        method : {None, 'pad'/'ffill', 'backfill'/'bfill', 'nearest'}, optional
             * default: exact matches only.
             * pad / ffill: find the PREVIOUS index value if no exact match.
             * backfill / bfill: use NEXT index value if no exact match
             * nearest: use the NEAREST index value if no exact match. Tied
               distances are broken by preferring the larger index value.
-        limit : int
-            Maximum number of consecuctive labels in ``target`` to match for
+        limit : int, optional
+            Maximum number of consecutive labels in ``target`` to match for
             inexact matches.
+        max_distance : optional
+            Maximum distance between target and index value for inexact
+            matches.
+
```

Comment:
I think you should raise somewhere if ``max_distance`` is not None and you have an incompatible ``method`` (e.g. right now only 'nearest')